### ISSUE TITLE
no-unknown-classes: named peer/group markers (peer/name, group/name) flagged as unknown while bare peer/group are exempt

### ISSUE THREAD

**Issue by esafak** [sergioazoc/oxlint-tailwindcss#102]
## Summary

In v1.5.0, `no-unknown-classes` reports Tailwind v4 **named peer/group marker classes** (`peer/menu-button`, `group/menu-item`, …) as unknown. The bare markers `peer` and `group` are already exempted by the precompute step with a comment noting they "don't produce CSS but enable group-hover:/peer-checked: variants". The named forms are the same construct and appear intended to be covered, but the handling is unimplemented.

## Environment

- `oxlint-tailwindcss`: **1.5.0** — passes on **1.4.0** with the same code (behavior change introduced by the exactness work)
- Tailwind CSS v4. Named peer/group markers are the documented v4 idiom for disambiguating multiple peers/groups, used pervasively by shadcn-ui (e.g. the sidebar component).

## Minimal repro

A class list containing a named marker, e.g. in a `tailwind-variants` template string:

```ts
import { tv } from "tailwind-variants";

export const sidebarMenuButton = tv({
  base: "peer/menu-button flex w-full items-center gap-2 rounded-md p-2",
});
```

Run with the default `no-unknown-classes` rule (no special options).

## Actual

```
tailwindcss(no-unknown-classes): "peer/menu-button" is not a valid Tailwind class.
```

Same for `group/<name>`.

## Expected

No diagnostic. `peer/menu-button` is a valid named-peer marker: intentionally CSS-less on its own (identical to bare `peer`/`group`), but **load-bearing** — sibling utilities consume it and compile to real CSS that references the marker. For example `peer-data-[size=sm]/menu-button:top-1` compiles to:

```css
.peer-data-\[size\=sm\]\/menu-button\:top-1:is(:where(.peer\/menu-button)[data-size="sm"] ~ *) {
  top: var(--spacing);
}
```

Removing the marker silently breaks those consumers, so it is not dead markup.

## Root cause

In the precompute worker:

- Bare `group`/`peer` are pushed into the valid set when a `group`/`peer` variant exists (block commented "Marker classes: group/peer don't produce CSS but enable group-hover:/peer-checked: variants").
- Immediately after, the comment "Named groups/peers: group/name, peer/name — the /name part is user-defined. These are validated by the variant system, not by candidatesToCss" is followed by **no implementation** — no named form is added to the valid set.

As a result, in the rule itself:

- `isKnownClass("peer/menu-button")` is `false`, so the class is sent to `candidatesToCss`, which returns `null` (the marker emits no CSS by design).
- The rule's own `classValidity("peer/menu-button")` returns `"valid"` — the slash-base fallback recognizes the base `peer` — but the strict gate `compiles === false` overrides that and reports.

So the rule is internally inconsistent: its classifier says the marker is valid, and only the layered "does it produce CSS?" check rejects it.

## Suggested fix

Exempt named marker forms the same way bare ones are exempted — when a `group`/`peer` variant exists, also treat any class matching `^(?:peer|group)/` (and prefixed forms) as a known marker so it is not sent to `candidatesToCss`. This matches the intent expressed in the existing comment and mirrors how bare `peer`/`group` are handled.

## Workaround

The `allowlist` option works:

```jsonc
{
  "rules": {
    "tailwindcss/no-unknown-classes": ["error", {
      "allowlist": ["peer/menu-button", "group/menu-item"]
    }]
  }
}
```

So this is a correctness/quality-of-life issue, not a hard blocker.

**Comment by sergioazoc** [comment]
Thanks for this. Reports that arrive with the root cause already traced are rare, and this one saved me real time.

Fixed in v1.7.0, just published.

You had it right except for where the fix goes. The precompute can't seed named markers: the `/name` is yours and unbounded, and `ds.parseCandidate('peer/menu-button')` returns `[]`, so there is nothing to enumerate. "When a group/peer variant exists" isn't checkable at lint time either, `cache.variantNames()` has 71 entries and neither `group` nor `peer` is among them (the precompute filters arbitrary variants, and Tailwind reports both as arbitrary).

So it lives in the rule now, behind a derived predicate: a precomputed, non-component class that emits zero declarations. Across every fixture that resolves to exactly `group` and `peer`, 2 of ~23,600 classes. No hardcoded names, so it self-prunes if Tailwind ever drops the variants, picks up any new CSS-less marker for free, and leaves `@container/main` alone because that one does have declarations.

More was broken than your repro showed: `!peer/menu-button`, `peer/menu-button!`, `hover:group/menu-item` and the prefixed `tw:peer/menu-button` were all reported too. It splits at the first slash now, so `peer//x` and `group/a/b` pass (Tailwind takes any non-empty name and never checks that it exists). `peer/` still reports, that one genuinely compiles to nothing.

Heads up on one thing: it wasn't report-only. For short names like `peer/a` the rule offered a quick-fix rewriting it to `peer`, which drops the marker name and leaves every `peer-*/a:` consumer matching nothing. Worth a grep if you applied any of those in an editor.

Your report also paid for itself twice. Auditing it turned up that the component-class scan was harvesting identifiers from Tailwind's own preflight comments, so every project has been accepting `com`, `org`, `css` and `5rem` as valid Tailwind classes. Also fixed in 1.7.0.

You can drop the allowlist now.
